### PR TITLE
Add docs page about plans for dropping py27 and py34

### LINKED ISSUE #4635: Add a page to the docs about our plans to drop Python 2.7 and 3.4

### PR DESCRIPTION
**Pull request by nicoddemus**
Fix #4635

### CODE REVIEW

**Comment by nicoddemus** [comment]
Hi @KengoTODA,

Shouldn't the rtd-bot post a link on this PR to the built documentation? 🤔

**Comment by nicoddemus** [comment]
Ahh I figured as much, thanks. :+1:

File: doc/en/py27-py34-deprecation.rst [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+will install the last compatible pytest ``4.X`` version automatically even if ``5.0`` or later
+are available on PyPI.
+
+During the period **from mid-2019 and 2020**, the pytest core team plans to make
+bug-fix releases of the pytest ``4.X`` series by back-porting patches to the ``4.x-maintenance``
+branch.
+
+**After 2020**, the core team will no longer actively back port-patches, but the ``4.x-maintenance``
+branch will continue to exist so the community itself can contribute patches. The
+core team will be happy to accept those patches and make new ``4.X`` releases **until mid-2020**.
```
[the remark this answers is not in the document]
Comment (RonnyPfannschmidt) [reply doc/en/py27-py34-deprecation.rst]: @nicoddemus the devil on my shoulder just wants to make the deprecation warning more and more passive aggressive with each release - as long s the release pipeline is accessible the effort to cut a release can be downsized to something nicely manageable

[the remark this answers is not in the document]
Comment (nicoddemus) [reply doc/en/py27-py34-deprecation.rst]: No problem with the warnings, I wonder though if we want to declare a final date where we are no longer obligated to make 4.X releases anymore (even if we still do occasionally)

[the remark this answers is not in the document]
Comment (RonnyPfannschmidt) [reply doc/en/py27-py34-deprecation.rst]: i believe we have about 9 months of active support (2.7 gone form mainline vs 2.7 eol), i think having a 6 month grace period after that to do releases from our side is fine, afterwards its the end of line from our side, no more releases but we should still merge to the maintenance branch in good faith (people have to manage their setups on their own then)

[the remark this answers is not in the document]
Comment (nicoddemus) [reply doc/en/py27-py34-deprecation.rst]: Agreed.

OK to merge this then?

**Pull request merged**